### PR TITLE
added restrictions for specific roles and updated account page

### PR DESCRIPTION
**Pull request by FiniDG**
new features
- fixes issue #3 
- new roles ROLE_CLINIC and ROLE_DIAGGEN added
- make restrictions for ROLE_CLINIC that they can never access the "Genes", "Variants" and "List All" pages
- make restrictions for ROLE_CLINIC that they cannot link via another page to "Genes", "Variants" and "List All" pages
- updated the "Account" page to let the user know about these new restrictions (you are always redirected to this page if access is denied)
- restrictions also apply for users that are not ROLE_DIAGGEN or ROLE_ADMIN

Additionally 
- updated some code to remove old comments and some formatting
- fixed issues in main.py addressed by reviewer of last pull request (feature/layout_changed) [link to discussion pull request](https://github.com/UMCUGenetics/VUSualizer/pull/13#discussion_r837277971).

### CODE REVIEW

File: src/templates/account.html [hunk 1]
```diff
@@ -3,29 +3,31 @@
...
-            </div>
+    <!-- Content Row -->
+    <div class="row">
+        <div class="col">
+            You are currently logged in as {{ name }},
+            {% if not current_user.active %}
+            however your account is not active yet.
+            <p>Therefore you cannot access the data. For more information contact bioinformatica-genetica@umcutrecht.nl.</p>
+            <p><a class="btn btn-primary" href="{{ url_for('logout') }}">logout</a> <a class="btn btn-default"
+                    href="{{ url_for('register') }}">register</a></p>
```
Comment (rernst) [inline src/templates/account.html:22]: Why a register button for logged in users?

File: src/templates/account.html [hunk 2]
```diff
@@ -3,29 +3,31 @@
...
 
-                <p>
-                <a class="btn btn-primary" href="{{ url_for('logout') }}">logout</a> <a class="btn btn-default" href="{{ url_for('register') }}">register</a>
-                </p>
-                {% endif %}
-            </div>
+    <!-- Content Row -->
+    <div class="row">
+        <div class="col">
+            You are currently logged in as {{ name }},
```
Comment (rernst) [inline src/templates/account.html:17]: This will always display a ','.

File: src/templates/account.html [hunk 3]
```diff
@@ -3,29 +3,31 @@
...
+            {% if not current_user.active %}
+            however your account is not active yet.
+            <p>Therefore you cannot access the data. For more information contact bioinformatica-genetica@umcutrecht.nl.</p>
+            <p><a class="btn btn-primary" href="{{ url_for('logout') }}">logout</a> <a class="btn btn-default"
+                    href="{{ url_for('register') }}">register</a></p>
+            {% elif current_user.role != "ROLE_DIAGGEN" and current_user.role != "ROLE_ADMIN" %}
```
Comment (rernst) [inline src/templates/account.html:23]: I prefer, it is easier to change / add roles.
```
if current_user.role not in ['ROLE_DIAGGEN', 'ROLE_ADMIN'] 
```

File: src/views/main.py [hunk 4]
```diff
@@ -22,7 +22,17 @@ def login_required(f):
...
+
+
+def diaggen_role_required(f):
+    '''Wrapper around the login_required() function of Flask, to also check the active status of the user'''
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        if current_user.role != "ROLE_ADMIN" and current_user.role != "ROLE_DIAGGEN":
             return redirect(url_for('account'))
         return f(*args, **kwargs)
     return decorated_function
```
Comment (rernst) [inline src/views/main.py:38]: - See comment in account.html
- Maybe add role as a parameter? Like: https://github.com/mattupstate/flask-security/blob/674b18103fa8734aca71bbd084ea01e3709817ef/flask_security/decorators.py#L174
- To much for this review, but this module https://flask-security-too.readthedocs.io/en/stable/ has all these role functions build in.

**Review by rernst — COMMENTED** [review]
Just some small remarks.

File: src/templates/account.html (hunk 2 above)
Comment (FiniDG) [reply src/templates/account.html:17]: but normally you are never in the "account" page, if you are not logged in (unless you type it directly in the web adres). 
The whole sentence will always be visible, not just the ",".
What is your suggestion here?

File: src/templates/account.html (hunk 1 above)
Comment (FiniDG) [reply src/templates/account.html:22]: yeah, that is kinda weird. I will change it

File: src/templates/account.html (hunk 3 above)
Comment (FiniDG) [reply src/templates/account.html:23]: agreed. I will change this (looks much better)

File: src/templates/account.html (hunk 2 above)
Comment (FiniDG) [reply src/templates/account.html:17]: I see what you mean now
```
Account
You are currently logged in as 123_diaggen@123.com,
```
I will just remove the ",". is not really necessary

**Review by rernst — APPROVED** [review]

**Pull request merged**